### PR TITLE
Migrate Test cases

### PR DESCRIPTION
**Pull request by tharikaGitHub**
This PR migrates the following test cases and fixes some other test failures.

1. ErrorResponseCheckTestCase
2. TagsRatingCommentTestCase
3. HttpPATCHSupportTestCase
4. EditAPIAndCheckUpdatedInformationTestCase

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml [hunk 1]
```diff
@@ -43,6 +43,7 @@
             <class name="org.wso2.am.integration.tests.oas.OASTestCase"/>
             <class name="org.wso2.am.integration.tests.api.lifecycle.APIVisibilityByPublicTestCase"/>
             <class name="org.wso2.am.integration.tests.api.lifecycle.APIVisibilityByDomainTestCase"/>
+            <class name="org.wso2.am.integration.tests.api.lifecycle.EditAPIAndCheckUpdatedInformationTestCase"/>
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml]: We are planning to include all the tests at once in the testng file. Shall we remove this for now?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIMANAGER4081PaginationCountTestCase.java [hunk 2]
```diff
@@ -189,6 +189,6 @@ public void destroy() throws Exception {
                 restAPIPublisher.deleteAPI(createdAPIs.get(j).getId());
             }
         }
-        super.cleanUp();
+        super.cleanUpUsingRest();
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIMANAGER4081PaginationCountTestCase.java]: Let's use the old cleanUp method as I have modified it to support latest rest api calls. Sorry for the confusion.

**Comment by mushthaq33** [comment]
Let's resolve the merge conflicts.

File: modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml (hunk 1 above)
Comment (tharikaGitHub) [reply modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml]: Fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIMANAGER4081PaginationCountTestCase.java (hunk 2 above)
Comment (tharikaGitHub) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIMANAGER4081PaginationCountTestCase.java]: Fixed

**Pull request merged**